README.md

```diff
@@ -128,52 +128,49 @@ A common scenario of processing actions is
 * Send message to Watson with updated context
 * Send result message(s) to user.
 
-#### Using middleware.after and controller
-Before v1.4.0 only middleware.after callback can update context, and only controller can send replies to user.
-The downside is that it is impossible to send "Please wait message".
-
+### using sendToWatson to update context (possible since v1.5.0)
+Using sendToWatson to update context simplifies the bot code compared to solution using updateContext below.
 ```js
-function checkBalance(watsonResponse, callback) {
-  //middleware.after function must pass a complete Watson respose to callback 
-  watsonResponse.context.validAccount = true;
-  watsonResponse.context.accountBalance = 95.33;
-  callback(null, watsonResponse);
+
+function checkBalance(context, callback) {
+  //do something real here
+
+  var contextDelta = {
+    validAccount: true,
+    accountBalance: 95.33
+  };
+  callback(null, context);
 }
 
-watsonMiddleware.after = function(message, watsonResponse, callback) {
-  //real action happens in middleware.after
-  if (typeof watsonResponse !== 'undefined' && typeof watsonResponse.output !== 'undefined') {
-    if (watsonResponse.output.action === 'check_balance') {
-      return checkBalance(watsonResponse, callback);
-    }
-  }
-  callback(null, watsonResponse);
-};
+Promise.promisifyAll(watsonMiddleware);
+var checkBalanceAsync = Promise.promisify(checkBalance);
 
-var processWatsonResponse = function(bot, message) {
+var processWatsonResponse = function (bot, message) {
   if (message.watsonError) {
     return bot.reply(message, "I'm sorry, but for technical reasons I can't respond to your message");
   }
-
   if (typeof message.watsonData.output !== 'undefined') {
     //send "Please wait" to users
     bot.reply(message, message.watsonData.output.text.join('\n'));
 
     if (message.watsonData.output.action === 'check_balance') {
       var newMessage = clone(message);
       newMessage.text = 'balance result';
-      //send to watson
-      watsonMiddleware.interpret(bot, newMessage, function() {
-        //send results to user
-        processWatsonResponse(bot, newMessage);
+
+      checkBalanceAsync(message.watsonData.context).then(function (contextDelta) {
+        return watsonMiddleware.sendToWatsonAsync(bot, newMessage, contextDelta);
+      }).catch(function (error) {
+        newMessage.watsonError = error;
+      }).then(function () {
+        return processWatsonResponse(bot, newMessage);
       });
     }
   }
 };
 
 controller.on('message_received', processWatsonResponse);
-
 ```
+
 #### Using updateContext in controller (available since v1.4.0)
```

Comment:
We may want to move code for specific versions out of the README and into a CHANGELOG wiki page..

Reply:
@Naktibalda you should be able to edit this page. https://github.com/watson-developer-cloud/botkit-middleware/wiki/Changelog Let's use it for users trying to find when something was fixed or how to update their code after we do a release.

Reply:
This isn't a "code for specific version", this is a new feature.
One reason to specify a version is because README becomes visible before release, so the documented feature isn't available yet.
I could remove the old ways of doing things, but someone could find them useful for doing something else.

Reply:
I think my comment was more toward having a CHANGELOG along with the README but not particular to this PR. I'm trying to remove old code from the README unless it's necessary. We keep changing the API and every time we do that we have to update the code snippet.